tgui/packages/tgui/interfaces/NtosPortraitPrinter.js

```diff
@@ -1,43 +1,83 @@
 import { resolveAsset } from '../assets';
 import { useBackend, useLocalState } from '../backend';
-import { Button, NoticeBox, Section, Stack } from '../components';
+import { Button, NoticeBox, Section, Stack, Input } from '../components';
 import { NtosWindow } from '../layouts';
 
 export const NtosPortraitPrinter = (props, context) => {
   const { act, data } = useBackend(context);
   const [listIndex, setListIndex] = useLocalState(context, 'listIndex', 0);
   const {
     paintings,
+    search_string,
+    search_mode,
   } = data;
-  const current_portrait_title = paintings[listIndex]["title"];
-  const current_portrait_asset_name = "paintings" + "_" + paintings[listIndex]["md5"];
+  const got_paintings = !!paintings.length;
+  const current_portrait_title = got_paintings
+    && paintings[listIndex]["title"];
+  const current_portrait_author = got_paintings
+    && "By " + paintings[listIndex]["creator"];
+  const current_portrait_asset_name = got_paintings
+    && "paintings" + "_" + paintings[listIndex]["md5"];
   return (
     <NtosWindow
       title="Art Galaxy"
       width={400}
-      height={406}>
+      height={446}>
       <NtosWindow.Content>
         <Stack vertical fill>
+          <Stack.Item>
+            <Section
+              title="Search">
+              <Input fluid
+                placeholder="Search Paintings..."
+                value={search_string}
+                onChange={(e, value) => {
+                  act('search', {
+                    to_search: value,
+                  });
+                  setListIndex(0);
+                }} />
+              <Button
+                content={search_mode}
+                onClick={() => {
+                  act('change_search_mode');
+                  if (search_string) {
+                    setListIndex(0);
+                  }
+                }} />
```

Comment:
This would probably be better as a drop-down menu that lets you choose the search type, no?

Reply:
I have considered that, but there are really only two modes right now so I didn't bother.